### ISSUE TITLE
More fine tune APK selection. 

### ISSUE THREAD

**Issue by S7venLights** [ImranR98/Obtainium#20]
Just installed Brave Browser with Obtanium. 
Issue is it will only select the newest APK, in this case it is Brave Nightly.
Ideally there'd be a way to see which apk is which.  Maybe we can select from a list that is displayed in web view.

**Comment by ImranR98** [comment]
I'm a bit confused about Brave's criteria for using the "Pre-release" tag. The latest full release (https://github.com/brave/brave-browser/releases/tag/v1.43.94) has the yellow "pre-release" tag, while some of the actual nightlies and betas don't (like https://github.com/brave/brave-browser/releases/tag/v1.45.54).

Right now, Obtainium picks the latest full release and ignores pre-releases (unless no full releases are available).

A toggle to control this is already planned (#18), but this uses the "pre-release" label so it won't work right with Brave if they don't label things correctly (or maybe there's some other logic behind their labelling that I'm missing?).

**Comment by S7venLights** [comment]
Hmm, in that case is it possible to give the user the ability to see more detail about the APK in the APK picker popup?  Perhaps show the tag and description?

**Comment by S7venLights** [comment]
Related (#22)

**Comment by ImranR98** [comment]
Regarding manual APK selection, that already exists for multiple APKs within a single release. The problem here is deciding which release to pick, and this needs to happen every time the app checks for updates. Meaning a 'prerelease/no-prerelease' toggle can only be set when the app is added or when the user manually changes it - we can't ask to choose that every time the app checks for updates.

**Comment by ImranR98** [comment]
Latest version allows you to add a regular expression to filter releases by name: https://github.com/ImranR98/Obtainium/releases/tag/v0.3.0-beta
Tested with Brave browser. The following pattern should work: `^Release`

**Comment by S7venLights** [comment]
Oh I missed this before, how does it work, sorry I don't follow?

**Comment by ImranR98** [comment]
When adding an app (only GitHub Apps for now), you can use the text box at the bottom ("Filter Release Titles...") to ignore any releases whose titles/names do not match a specific text pattern. So if the latest release title is `Nightly v1.46.8` but you instead want `Release v1.44.97`, you can set the pattern to `Release` so it only grabs the latest release whose title containes that word. This pattern can also be changed later on from the App page.

99% of the time this will not be needed to filter out pre-releases as we could normally just use the "Include Prereleases" toggle but that doesn't work for Brave since they don't label releases correctly, so we can filter by release titles instead.

The text box has [Regular Expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) support for more advanced filtering, but you don't need to know what that is just for basic use like in the example above.

**Comment by S7venLights** [comment]
Thanks, I understand, and it worked. Now what would someone do if they had nightly and regular release installed and they wanted to update both? 
Remove the regular expression in between updates?

**Comment by S7venLights** [comment]
Doesn't seem to be working for the latest version, know why?

**Comment by ImranR98** [comment]
![Screenshot_20221008-170050](https://user-images.githubusercontent.com/30463115/194727685-a0455503-02ab-4463-b89e-4ef17d6fed21.png)
These settings give me the latest release version.

**Comment by ImranR98** [comment]
> Thanks, I understand, and it worked. Now what would someone do if they had nightly and regular release installed and they wanted to update both? Remove the regular expression in between updates?

This is not doable right now, and I'm not sure if it's worth looking into just to support that niche use case. Because App IDs are generated from their URLs and are meant to be unique, so you can't have 2 instances of the same URL. Supporting this would get even more confusing since the user can modify their regular expression filter at any time, so it's not a constant that can be made part of the unique ID.

**Comment by ImranR98** [comment]
@S7venLights I just took a look at your export JSON. Brave browser isn't working for you because they mark their full releases as 'pre-release', and you have toggled off the "include prereleases" option.

**Comment by S7venLights** [comment]
I suppose if I had nightly and standard release I could change the regular expression each time I update. But I no longer have nightly installed anyway.

**Comment by S7venLights** [comment]
So lately Brave һаѕո't released a new "release" version, only nightly & Beta. 
Small issue, but despite using the the 'Release' regular expression, Obtainium keeps telling me: couldn't find suitable release. 
My guess is it's not looking far back in the releases page to find the release.

**Comment by S7venLights** [comment]
@ImranR98 
Okay, Brave has a new release as of 30 odd minutes ago and Obtanium is not detecting it. I have Include pre-release, fallback to older releases and Regular expression ^Release set. 

Brave was set as uninstalled in Obtanium although it is installed and during update checks it sends an error: Could not find a suitable release. 

I just removed it from Obtanium and attempted to re-add it but it got the same error and failed to add the app.

**Comment by ImranR98** [comment]
Changed something recently that I shouldn't have. Fixed: https://github.com/ImranR98/Obtainium/releases/tag/v0.6.9-beta

**Comment by S7venLights** [comment]
🙏

**Comment by netizeni** [comment]
Hi @ImranR98 ,

It's been a while since this issue was solved, so I want to check if something changed meantime, because Obtainium doesn't pull the latest version of Brave on my device.

Obtainium is on the latest version, and I've tried what's suggested here and played a bit with settings, i.e. trying both enabled and disabled options. At the moment of this writing, it pulls "v1.63.161 Latest" version when `Release` or `^Release` is added in filter titles, yet the latest version is "v1.63.162" which can be verified here: https://github.com/brave/brave-browser/releases/latest 

Both versions have `Release` in the title, while "v1.63.162" also has "Latest" tag. Is there something additionally which should be setup now?

**Comment by RasheedAZ** [comment]
> v1.63.162

I just noticed that this version doesn't have an apk... so perhaps no change Android version?

**Comment by ImranR98** [comment]
Yeah if there's no APK, Obtainium will just fallback to annolder release (if that is enabled). No idea why they haven't attached any APKs.

**Comment by netizeni** [comment]
Yeah, I noticed the same thing yesterday when I started looking for the solution, just haven't had a chance to write a comment before.

It's strange they didn't attach any apk, yet they updated Chromium version.
